### ISSUE TITLE
option groups/aliases

### ISSUE THREAD

**Issue by rbonthond** [BrianPugh/cyclopts#19]
would be nice if we could group options and add aliases for subcommands, like done with https://github.com/janluke/cloup

**Comment by BrianPugh** [comment]
I can certainly add aliases for commands like:

```
@app.command(name=["foo", "bar"])  # Register the function ``foo`` to CLI tokens "foo" and "bar"
def foo():
    pass
```

Option groups are hard. I think the API would have to look something like:

```
from cyclopts import App, Group, Parameter

app = App()

cool_options = Group(name="Cool options", mutually_exclusive=True)

@app.command
def foo(
    *,
    bar: Annotated[Optional[str], Parameter(group=cool_options)] = None,
    baz: Annotated[Optional[str], Parameter(group=cool_options)] = None,
):
    pass
```

What are your thoughts?

**Comment by BrianPugh** [comment]
I mostly implemented the alias feature (see `command-alias` branch), but actually, I think it might be better to just have the user multiple decorate their function (which the existing library code already achieves):

```
@command(name="foo")
@command(name="bar")
def blah(): 
    pass
```

This achieves the same thing without complicating the code more. The main argument I see for allowing a list for `name` is that it's more consistent with how `Parameter` works.

**Comment by rbonthond** [comment]
multiple commands would also work. might get a bit unreadable though perhaps?

**Comment by BrianPugh** [comment]
I think I'll continue adding this feature; I think maintaining consistency with all the other functionality (lots of parameters in `Parameter` can be a string or a list) is very important.

**Comment by BrianPugh** [comment]
command aliases are now available in v1.1.0.

Keeping this issue open to discuss the best way to handle groups.

**Comment by BrianPugh** [comment]
Closing this issue to centralize discussion over to #39.
